src/js/controller/controller.js

```diff
@@ -229,13 +224,13 @@ Object.assign(Controller.prototype, {
                 _preplay = false;
             }
 
-            _checkAutoStart();
-
             _model.change('viewable', viewableChange);
             _model.change('viewable', _checkPlayOnViewable);
             _model.once('change:autostartFailed change:autostartMuted change:mute', function(model) {
                 model.off('change:viewable', _checkPlayOnViewable);
             });
+
+            _checkAutoStart();
```

Comment:
Run `_checkAutoStart()` last since 'viewable' changes can result in `preload()` being called on the initial provider instance.

Reply:
Worthy of a comment